docs: Add SAML service provider with entity id and acs url

flyinghermit opened the pull request:
Add service provider with entity id and acs url. Make entity descriptor configuration optional.

Also shuffles the steps as explained below.

Previous:
```
1. Configure role
2. Download Teleport metadata and upload to Service Provider
3. Add Service Provider to Teleport
4. Test
```

I think these steps require user to move back and forth, first start with web browser, then `tctl` and then back to web browser. I think moving step 3 as 1 would enable working with `tctl` once and then configure everything in browser. The first step `Configure role` is now moved to prerequisites as optional step. Referencing preset `editor` isn't bad for a quickstart and users would create dedicated role for production. 
 
So now the steps are:
```
Prerequisites:
...
- Configure role

Steps:
1. Add Service Provider to Teleport
2. Download Teleport metadata and upload to Service Provider
3. Test
```

flyinghermit committed:
add service provider with entity id and acs url. make entity descriptor optional config

zmb3 commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -15,69 +15,98 @@ authenticate to external services.
-create a role called `sp-manager.yaml` with the following contents:
-
+  Reference](./saml-reference.mdx) before proceeding.
+- User with preset `editor` role. We recommend creating a dedicated role to manage
```
This point is a little bit confusing to me. We list the preset editor role as a perquisite (which means you must have it in order to continue), but then we immediately say "we recommend creating a dedicated role [and not using the preset]".

Can we be more explicit as to what we recommend?

flyinghermit replied:
Move the role creation block to the end of the guide https://github.com/gravitational/teleport/pull/36467/commits/20806d4d4e6d30df09e5ec9f756d329f988623b7

This should address your concern here and below.

zmb3 commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -15,69 +15,98 @@ authenticate to external services.
+Service Provider.
+
+  <Admonition type="tip" title="Recommended">
+  <Details title="Create SAML IdP Service Provider role" closed>
```
Not sure what others thought are, but I've always preferred information-dense text as opposed to fancy UI controls or other "noise" on the page. Putting a collapsible box inside an admonition feels like a lot of cognitive load. Can we simplify?

zmb3 commented on docs/pages/access-controls/idps/saml-guide.mdx:
````diff
@@ -15,69 +15,98 @@ authenticate to external services.
+  ```
+  </Details>
+  </Admonition>
+- Test SAML application, also known as SAML Service Provider (SP). For this guide, we'll be using 
````
This is also a bit tricky. This bullet actually corresponds to the list above, where the previous list item is 30 lines away. Can you think of ways to restructure this so we don't break the user's context? By the time I got to this point I forgot that I was still reading preqrequisites.

flyinghermit replied:
Hmmm, I get your concern, I assume that is the case while reading directly from the `mdx` file and not through rendered page?  Though the block above is now moved to end (https://github.com/gravitational/teleport/pull/36467/commits/20806d4d4e6d30df09e5ec9f756d329f988623b7) so should address your concern here.

flyinghermit commented:
Note to reviewer, testing with samltest.id is broken currently and will be resolved with https://github.com/gravitational/teleport/pull/36057

flyinghermit committed:
move dedicate role creation to end of the guide

mdwn commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -127,3 +122,47 @@ This has verified service provider initiated SSO. To verify identity provider in
+
+For production, we recommend creating a dedicated role to manage Service Provider.
+
+To create dedicated role, first, ensure you are logged into Teleport as a user that has
```
```suggestion
To create a dedicated role, first, ensure you are logged into Teleport as a user that has
```

nklaassen commented on docs/pages/access-controls/idps/saml-guide.mdx:
````diff
@@ -127,3 +122,47 @@ This has verified service provider initiated SSO. To verify identity provider in
+role 'saml-idp-service-provider-manager' has been created
+```
+
+Add the role to your user. Replace `teleport-admin` with your user name:
+
+```code
+$ tctl users update --set-roles $(tctl get users/<Var name="teleport-admin"/> --format=json | \
+jq -r '.[].spec.roles | join(",")'), sp-manager <Var name="teleport-admin"/>
+User teleport-admin has been updated:
+  New roles: auditor,editor,access,sp-manager
+```
````
This only works with local users, consider referencing docs/pages/inclues/add-role-to-user.mdx

flyinghermit replied:
Thanks for that. I didn't figure we have that in includes. Updated in https://github.com/gravitational/teleport/pull/36467/commits/946f3ed6d5c1c5779f9bafc12252e6c208a29dc6 along with your other suggestions above.

flyinghermit committed:
review comments:
- copy edits
- use add-role-to-user.mdx to reference role update

flyinghermit commented:
Friendly ping.

flyinghermit committed:
reference content of entity descrptor takes priority over other values

flyinghermit commented:
Friendly ping @r0mant @ptgott @nklaassen @zmb3

ptgott commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -15,69 +15,53 @@ authenticate to external services.
 - If you're new to SAML, consider reviewing our [SAML Identity Provider
-Reference](./saml-reference.mdx) before proceeding.
+  Reference](./saml-reference.mdx) before proceeding.
+- User with permission to create Service Provider resource. The preset `editor` role has this permission.
```
We should explain what a Service Provide resource is

flyinghermit replied:
The glossary for SAML IdP is published on this page https://goteleport.com/docs/access-controls/idps/saml-reference/, which also covers service provider, and the reference page is linked just before this line.

ptgott commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -15,69 +15,53 @@ authenticate to external services.
-Reference](./saml-reference.mdx) before proceeding.
+  Reference](./saml-reference.mdx) before proceeding.
+- User with permission to create Service Provider resource. The preset `editor` role has this permission.
+- Test SAML application, also known as SAML Service Provider (SP). For this guide, we'll be using
```
What does "also known as SAML Service Provider (SP)" mean here? Does this rename "Test SAML application"?

flyinghermit replied:
Not really, but I updated this line so to avoid such confusion:
`SAML application (also known as a SAML service provider or SP) for testing. For this guide, we'll be using
[samltest.id](https://samltest.id/) as a service provider for Teleport's SAML IdP.`

ptgott commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -15,69 +15,53 @@ authenticate to external services.
+[samltest.id](https://samltest.id/) to create a test consumer of Teleport's SAML IdP.
 
-## Example external application
+## Step 1/3. Add samltest.id Service Provider to Teleport
```
```suggestion
## Step 1/3. Add a Service Provider to Teleport
```
Using more general headings so users can create a mental model of the flow. Otherwise, I think this is too specific to the example we're using.

flyinghermit replied:
Totally. I borrowed the copy from existing page but I felt the same way. updated this and your other suggestions in https://github.com/gravitational/teleport/pull/36467/commits/7193225bab9b229e414415717a268093a129f76f

ptgott commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -15,69 +15,53 @@ authenticate to external services.
+kind: saml_idp_service_provider
 metadata:
-  name: sp-manager
+  # The friendly name of the service provider. This is used to manage the
```
We should capitalize "Service Provider" consistently. Using lowercase everywhere probably makes more sense, since this is a general computing term rather than a Teleport product.

ptgott commented on docs/pages/access-controls/idps/saml-guide.mdx:
```diff
@@ -106,10 +90,31 @@ Add this to Teleport using `tctl`:
+</Admonition>
 
-Teleport now trusts the samltest.id service provider.
+## Step 2/3. Configure samltest.id to recognize Teleport's SAML IdP
```
```suggestion
## Step 2/3. Configure the Service Provider to recognize Teleport's SAML IdP
```

flyinghermit committed:
copy edits:
- use lower case for service provider
- use generic service provider name instead of directly referencing samltest.id
- update test saml application copy

flyinghermit committed:
fix typo
